app/src/org/commcare/activities/SyncCapableCommCareActivity.java

```diff
@@ -75,15 +75,12 @@ public void onAttachedToWindow() {
      * triggered after they are submitted. If no forms are sent, triggers a sync explicitly.
      */
     protected void sendFormsOrSync(boolean userTriggeredSync) {
-        boolean formsSentToServer = checkAndStartUnsentFormsTask(true, userTriggeredSync);
-        if (!formsSentToServer) {
```

Comment:
we now return a success result from upload forms task in case the # of forms to upload is 0 which in turn triggers the data pull.